src/main/java/io/electrum/billpay/validation/ConsistentAdviceIds.java

```diff
@@ -0,0 +1,35 @@
+package io.electrum.billpay.validation;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+/**
+ * Checks that for resources which have the transaction identifier and the request transaction identifier as a path
+ * parameter (resulting in 9 parameters in total), where the first three parameters are (String tranId, String
+ * requestId, (? extends Transaction) entity), that the tranId is the same as the identifier in the BasicAdvice, and
+ * that the requestId is the same as the request identifier in the BasicAdvice.
+ */
```

Comment:
I think that in javadoc, especially public facing javadoc, it's useful to link back to the class you mention:
```suggestion
/**
 * Checks that for resources which have the transaction identifier and the request transaction identifier as a path
 * parameter (resulting in 9 parameters in total), where the first three parameters are {@code (String tranId, String
 * requestId, (? extends {@link Transaction}) entity)}, that the tranId is the same as the identifier in the {@link BasicAdvice}, and
 * that the requestId is the same as the request identifier in the {@link BasicAdvice}.
 */
```